### PR TITLE
[Ember-Migration] Created a navbar

### LINKED ISSUE #288: [Ember-Migration] Create a Navbar similar to current main-site
- Create a navbar
- Add a dropdown on clicking the user-greeting

### PR DESCRIPTION
**Pull request by rohan09-raj**
## Resolves Issue
Closes #288 

## What is the change?
- Create a navbar 

## Is Development Tested?
- [x] Yes

## Before / After Change Screenshots
image

image

image

image

## Note for the Reviewer
- CI is failing due to package-lock.json being not found as we are using yarn.lock here. Therefore, need to update the Workflows or create a new workflow. Issue for the same - #327 
- SignIn Sign Up flow is not completed yet. It will be handled separately #326

### CODE REVIEW

File: app/components/navbar.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    },
+    {
+      siteName: 'Crypto',
+      url: APPS.CRYPTO,
+    },
+    {
+      siteName: 'Status',
+      url: APPS.STATUS,
+    },
+  ];
```
Comment (prakashchoudhary07) [inline app/components/navbar.js:34]: Can we move this to constant file?

File: app/components/navbar.hbs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      </a>
+    </li>
+    <li>
+      <a class='nav__home__link nav__element' href={{this.HOME_PAGE_URL}}>
+        Home
+      </a>
+    </li>
+    {{#each this.RDS_APPS_MAPPING as |siteURL|}}
+    <li><a class='nav__element' href={{siteURL.url}}>{{siteURL.siteName}}</a></li>
+    {{/each}}
```
Comment (prakashchoudhary07) [inline app/components/navbar.hbs:19]: Can we add home also from in #each

File: app/components/navbar.hbs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      this.outsideClickMenu}}>
+      <div class='user__msg'>Hello, {{@firstName}}</div>
+      <img class='user__pic' src={{@profilePicture}} />
+      <Fa-Icon @icon='chevron-down' />
+    </div>
+    {{else}}
+    <a class='login' href={{this.AUTH_URL}}>
+      <button class='login__btn' type="button">
+        <span class='login__text1'>Sign In</span>
+        <span class='login__text2'>with GitHub</span>
```
Comment (prakashchoudhary07) [inline app/components/navbar.hbs:34]: NIT: CSS naming could be different, text1 and text2 not descriptive

File: app/components/navbar.hbs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        <img class='login__logo' src='assets/icons/github-logo.png' alt='GitHub_Icon' height='20px' width='20px' />
+      </button>
+    </a>
+    {{/if}}
+  </div>
+
+  <section class="menu {{if this.isMenuOpen " active-menu" "" }}">
+    <a href="https://my.realdevsquad.com/" class="menu__link">
+      My Profile
+    </a>
```
Comment (prakashchoudhary07) [inline app/components/navbar.hbs:44]: NIT: link from constant

File: app/models/user.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import Model, { attr } from '@ember-data/model';
+
+export default class UserModel extends Model {
+  @attr('string', { defaultValue: 'User' }) first_name;
```
Comment (prakashchoudhary07) [inline app/models/user.js:4]: Why are we setting first_name default value to `User`

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: tests/integration/components/navbar-test.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+    // Template block usage:
+    await render(hbs`
+      <Navbar>
+        template block text
+      </Navbar>
+    `);
+
+    assert.dom(this.element).hasText('template block text');
+  });
```
Comment (prakashchoudhary07) [inline tests/integration/components/navbar-test.js:25]: Are we not writing tests?

File: app/components/navbar.hbs (hunk 2 above)
Comment (rohan09-raj) [reply app/components/navbar.hbs:19]: We can but need to extra condition for classes applied. 
I have removed the over simplification of this looping

File: app/components/navbar.hbs (hunk 4 above)
Comment (rohan09-raj) [reply app/components/navbar.hbs:44]: Done

File: app/components/navbar.hbs (hunk 3 above)
Comment (rohan09-raj) [reply app/components/navbar.hbs:34]: Done

File: app/models/user.js (hunk 5 above)
Comment (rohan09-raj) [reply app/models/user.js:4]: I think we can remove the default first_name

File: tests/integration/components/navbar-test.js (hunk 6 above)
Comment (rohan09-raj) [reply tests/integration/components/navbar-test.js:25]: I am thinking of covering all the tests in separate PRs and will assign to other people as well who wants to learn testing.

**Review by prakashchoudhary07 — APPROVED** [review]

**Comment by prakashchoudhary07** [comment]
NIT: .gitkeep within the controllers, route, and model can be removed, We have one on the root level, that should be enough I think

**Comment by prakashchoudhary07** [comment]
Why is build failing?

**Comment by rohan09-raj** [comment]
> NIT: .gitkeep within the controllers, route, and model can be removed, We have one on the root level, that should be enough I think

Will remove all the .gitkeep in another PR, as the code is still progressing and folders are being filled out with files

**Comment by rohan09-raj** [comment]
> Why is build failing?

It is due to the CI failing because of the old GitHub workflow. I have mentioned in the description also as a `Note for the reviewer

**Pull request merged**